### PR TITLE
Driver specific templates followup

### PR DESCRIPTION
**Pull request by alltilla**
Follow up for #4666.

Fixing https://github.com/syslog-ng/syslog-ng/pull/4666#discussion_r1365036653 and https://github.com/syslog-ng/syslog-ng/pull/4666#discussion_r1365085384.

### CODE REVIEW

File: lib/template/tests/test_template.c [hunk 1]
```diff
@@ -425,6 +425,10 @@ Test(template, test_escaping)
                                        FALSE, "\"value\"");
   assert_template_format_with_escaping("$(if (\"${APP.VALUE}\" eq \"value\") \"${APP.QVALUE}\" \"${APP.QVALUE}\")",
                                        TRUE, "\\\"value\\\"");
+
+  /* literal strings are not escaped */
+  assert_template_format_with_escaping("\"$(echo $(echo $(length ${APP.QVALUE})))\"", TRUE, "\"7\"");
```
Comment (bazsi) [inline lib/template/tests/test_template.c:430]: This is not a literal string, so I am not sure the comment matches the testcase.

**Review by bazsi — DISMISSED** [review]
Apart from a nitpick in a unit test this looks good to me.

File: lib/template/tests/test_template.c (hunk 1 above)
Comment (alltilla) [reply lib/template/tests/test_template.c:430]: I agree, this comment is misleading. What I wanted to check with these is that if there are tf/macro parts of the template, they are escaped, but the literal string parts are not (`"` in the beginning and the end).

I will rephrase this comment.

Comment (alltilla) [reply lib/template/tests/test_template.c:430]: `/* literal parts of the template are not escaped */`

Comment (bazsi) [reply lib/template/tests/test_template.c:430]: :) thanks

**Review by bazsi — APPROVED** [review]

**Pull request merged**